### DISCUSSION TITLE
LOF - File for Audacity - Does not support "Umlaute" and special characters öé

### DISCUSSION THREAD

**Discussion by reinhardwh** [jamulussoftware/jamulus#1384] [Bug Investigation]
Hello,

in case profile name contain "Umlaute" öäü or special characters like é also the WAV filename contains these characters when recording.

Within the LOF file also the name will be used. Example "Zuhörer"

Audacity cannot import files with special characters.

Is it possible to convert filenames and replace such characters using a-z 0-1 when recording.

**Comment by ann0see** [comment]
Maybe it's related to encoding? Can you open an issue on this topic please (best on the audacity repository).

**Comment by reinhardwh** [comment]
Hej ann0see,
just tested, encoding is ok ö = ö

Renamed my profile to Reinyö
-> LOF Datei cannot be imported to Audacity

Error Message Audacity
![jamulus-umlaut-1-3](https://user-images.githubusercontent.com/77500932/113199891-f08d2200-9267-11eb-9a79-a470fcfd6db4.jpg)

LOF Files created
![jamulus-umlaut-2-3](https://user-images.githubusercontent.com/77500932/113199906-f3881280-9267-11eb-9190-1524c2751cee.jpg)

LOF File content
![jamulus-umlaut-3-3](https://user-images.githubusercontent.com/77500932/113199913-f6830300-9267-11eb-9a99-e719761af7a7.jpg)

When removing the "ö" from filename and correcting LOF file the import just works fine....

**Reply by ann0see** [reply]
So it is an issue in Audacity in my opinion.

**Reply by reinhardwh** [reply]
Well, u r right.
Option 1: Jamulus not creating files containing "öä.."
or
Option 2: Audacity allow importing files including "öä.."

**Reply by ann0see** [reply]
> Jamulus not creating files containing "öä.."

For me that would mean a step backwards. My E-Mail provider does the same with passwords "for compatibility reasons" and personally I don't think that's the right approach. Can you contact audacity devs/open an issue on their repo?

Jamulus should use UTF8 (if it doesn't, we should probably try to add support).

**Reply by reinhardwh** [reply]
Hi ann0see,
will contact Audacity :)

**Reply by ann0see** [reply]
Great. Can you link the issue here?

**Comment by softins** [comment]
I wonder if this is an issue with UTF-8 vs ISO-8859-1?

**Reply by reinhardwh** [reply]
Hello ann0see,

Contacted Audacity as suggested:

https://forum.audacityteam.org/viewtopic.php?f=46&t=116997

**Reply by reinhardwh** [reply]
Reply from Audacity:

`
Currently LOF files are limited to standard ASCII text characters only.
(See: lower ASCII characters table)

Please note that it is generally recommended to use standard ASCII characters only for file names as the use of other characters will frequently cause compatibility problems if the files are shared with other people, particularly when sharing across networks.
`

**Reply by reinhardwh** [reply]
Post from Audacity

https://www.paho.org/par/index.php?option=com_docman&view=download&alias=560-guide-to-proper-file-naming&category_slug=organizacion-de-la-representacion&Itemid=253

**Reply by ann0see** [reply]
> Please note that it is generally recommended to use standard ASCII characters only for file names as the use of other characters will frequently cause compatibility problems if the files are shared with other people, particularly when sharing across networks.

Which will only get worse if everybody followed it. 
We're in 2021 and äöü éï etc. are supported by a lot of software. 

I understand your problem and probably there's no way around it except for re writing file names (although I think Audacity should fix it on their side; I hope they're doing it.)

**Reply by reinhardwh** [reply]
Hi ann0see

Agreed we are in 2021 : )

I asked my band member to change his profile name back to 1990 standard 

All good for now
Hopefully Audacity fix it soon

Happy easter

**Reply by ManfredSchenk** [reply]
I think there will be no fix in Audacity - when you want to write some software that is multi-platform (Windows, Linux, mac, iOS, Android, ...) you have to consider all limitations of those platforms. And when it comes to legal characters in filenames - ASCII is still the safest choice since you will get in trouble when allowing additional characters. Another solution might be to use some encoding of non-ASCII characters or BASE64-encode the whole filename before actually writing it - but then it becomes unreadable for humans.

**Reply by reinhardwh** [reply]
I like the "KISS" principle.

Maybe WIKI pages should mention to use "a-z;0-9" for profile names in Jamulus to avoid issues during recording/mixing later.

**Reply by gilgongo** [reply]
Or we could flag "illegal" use in the Alias/Name field of the Jamulus profile? I would think the number of people who read the docs are tiny :-)

**Reply by reinhardwh** [reply]
Hi gilgongo
Good idea. 
Or using one free profile name for display and another one for recording - restricted to a-z;0-1. 
The "recording" name code be auto-generated.

**Reply by gilgongo** [reply]
> The "recording" name code be auto-generated.

Yes that's probably best. Let the user input stand, and then fix it without bothering them about it.

Seeing as we have potential fix, should I raise an issue for this so that somebody could work on it?

**Reply by gilgongo** [reply]
OK done that https://github.com/jamulussoftware/jamulus/issues/1424

**Reply by reinhardwh** [reply]
Audacity Forum:
https://forum.audacityteam.org/viewtopic.php?f=46&t=116997
